### ISSUE TITLE
Not generating all fields from class when having children/sibling of same type

### ISSUE THREAD

**Issue by runarhk** [micronaut-projects/micronaut-openapi#442]
I have created a simple setup reproducing the where we have a class `a` that implements an interface `A` , having a field that has type `A` . 

This 

### Steps to Reproduce

1. Clone supplied repo
2. run gradle assemble
3. notice missing fields in type TestImpl1 in generated file at build/classes/java/main/META-INF/swagger/hello-world-0.0.yml

### Expected Behaviour

The field name `woopsie` should be in the TestImpl1 type with `type: $ref: '#/components/schemas/TestImpl1'`

### Actual Behaviour

The field `woopsie` is missing entirely. This is also the case for lists of same type. 

### Environment Information

- **Operating System**: Linux Mint
- **Micronaut Version:** TODO
- **JDK Version:** TODO

### Example Application

- TODO: link to github repository with example that reproduces the issue

**Comment by graemerocher** [comment]
The field `woopsie ` is private, why would it be included in the public schema?

**Comment by runarhk** [comment]
thanks for quick reply @graemerocher, i forgot to push the getter and setter which I added after removing lombok to get the test project as slim as possible. Pushed now ;)

**Comment by alvarosanchez** [comment]
This happens because when the `whoopsie` field is visited, the `TestInterface` type is also being visited in progress, and there is a recursion break to stop processing it.

Not sure what solution could be applied here.

**Comment by runarhk** [comment]
Makes sense. But how come it works if I add a generic type to the interface?

**Comment by alvarosanchez** [comment]
I guess that in that case, it goes through a different code path since `TestInterface<T>` != `TestInterface`
